unity-renderer/Assets/DCLPlugins/UUIDEventComponentsPlugin/UUIDComponent/OnPointerEvent.cs

```diff
@@ -109,6 +115,8 @@ public override void Initialize(IParcelScene scene, IDCLEntity entity)
 
             entity.OnShapeUpdated -= SetEventColliders;
             entity.OnShapeUpdated += SetEventColliders;
+            
+            DataStore.i.sceneBoundariesChecker.Add(entity,this);
```

Comment:
Maybe it would be nice to inject the `DataStore` as a dependency in this `Initialize()` method.

Reply:
You are right that sounds good, but that would require a big refactor as we should change the `Initialize()` method from [`IEntityComponent`](https://github.com/decentraland/unity-renderer/blob/dev/unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/Interfaces/IEntityComponent.cs), as the `Initialize()` of the `OnPointerEvent` is called in [`ECSComponentsManagerLegacy.EntityComponentCreateOrUpdate()`](https://github.com/decentraland/unity-renderer/blob/dev/unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/ECSComponentManagerLegacy/ECSComponentsManagerLegacy.cs#L421) along with the `Initialize()` of every other component implementing the `IEntityComponent` interface.